### PR TITLE
Mec2016evb port

### PR DESCRIPTION
**Pull request by franciscomunoz**
Initial commit for Microchip evaluation board

### CODE REVIEW

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst [hunk 1]
```diff
@@ -0,0 +1,212 @@
+.. _mec2016evb_assy6797:
+
+MEC2616EVB_ASSY6797
+###################
+
+Overview
+********
+
+The MEC2016EVB_ASSY6797 kit is a development platform to evaluate the
+Microchip MEC1701X series microcontrollers. This board needs to be matted with
```
Comment (dbkinder) [inline boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: I suspect this should this be ``mated`` (unless you mean is it "tangled into a thick mass" :)

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+The default configuration can be found in the Kconfig file:
+
+        ``boards/arm/mec2016evb_assy6797/mec2016evb_assy6797_defconfig``
+
+
+Connections and IOs
+===================
+
+Microchip to provide the schematic for this board.
```
Comment (dbkinder) [inline boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: Is there a link you can refer to here?

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+System Clock
+============
+
+The MEC1701 MCU is configured to use the 48Mhz internal oscilator with the
+on-chip PLL to generate a resulting EC clock rate of 12 MHz. See Processor clock
+control register (chapter 4 in user manual)
```
Comment (dbkinder) [inline boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: Is there a link to the user manual you can add here?

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+Serial Port
+===========
+
+UART0 is configured for serial logs.
+
+Jumper settings
+***************
+
+Please follow the jumper settings below for in order to properly demo this
```
Comment (dbkinder) [inline boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]:
```suggestion
Please follow the jumper settings below to properly demo this
```

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+| 1-2   | 1-2  | 1-2  | 1-2  | 1-2  | 1-2  | 1-2   | 1-2   |
++-------+------+------+------+------+------+-------+-------+
+
++-------+-------+------+------+------+-------+
+| JP56  | JP57  | JP58 | JP60 | JP61 | JP102 |
++=======+=======+======+======+======+=======+
+| 1-2   | 1-2   | 1-2  | 1-2  | 1-2  | 2-3   |
++-------+-------+------+------+------+-------+
+
+This jumpers configure VCC Powergood, nRESETI and JTAG_STRAP respectively.
```
Comment (dbkinder) [inline boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: It appears that ``power good`` should be two words:
```suggestion
These jumpers configure VCC Power good, nRESETI, and JTAG_STRAP respectively.
```

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
++-------+-------+------+
+| JP1   | JP2   | JP51 |
```
Comment (dbkinder) [inline boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: consider adding the headings you just mentioned into the table:
```
+-------------------+-------------+--------------+
| JP1               | JP2         | JP51         |
| (VCC Power good)  | (nRESETI)   | (JTAG_STRAP) |
+===================+=============+==============+
| 1-2               | 1-2         | 2-3          |
+-------------------+-------------+--------------+
```

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
++=======+=======+======+
+| 1-2   | 1-2   | 2-3  |
++-------+-------+------+
+
+Each column of the following table illustrates how to enable UART0, JTAG,
+PVT SPI, SHD SPI and LED0-3 respectively.
+
++-------+-------+---------+--------+--------+
+| JP27  | JP10  | JP34    | JP75   | JP68   |
```
Comment (dbkinder) [inline boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: Same comment here (add the names you mention above to this table heading)

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
++-------+-------+---------+--------+--------+
+|       |       | 14-15   | 14-15  |        |
++-------+-------+---------+--------+--------+
+|       |       | 17-18   | 17-18  |        |
++-------------------------------------------+
+
+Jump settings for MEC170x 144WFBGA Socket DC Assy 6801 Rev B1p0
+===============================================================
+
+The jumper configuration explained above covers the base board. Now the cpu
```
Comment (dbkinder) [inline boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]:
```suggestion
The jumper configuration explained above covers the base board. Now the CPU
```

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
++=======+=======+
+| 1-2   | 2-3   |
++-------+-------+
+
+Programming and Debugging
+*************************
+
+This board comes with a Cortex ETM port which facilitates tracing and debugging
+using a single physical conection.  In attidion, it comes with sockets for
+Jtag only sessions.
```
Comment (dbkinder) [inline boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]:
```suggestion
JTAG only sessions.
```

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+   string. Connection should be configured as follows:
+
+   - Speed: 115200
+   - Data: 8 bits
+   - Parity: None
+   - Stop bits: 1
+
+#. Connect the MEC2016EVB_ASSY_6797 board to your host computer using the
+   UART0 port. Then build :ref: `hello_world` application. It is important
+   to generate a binary with a new load address, for example do the following:
```
Comment (dbkinder) [inline boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: The easy way to get a code-block is to use a double colon at the end of the sentence:
```suggestion
   to generate a binary with a new load address, for example do the following::
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst (hunk 1 above)
Comment (franciscomunoz) [reply boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: done

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst (hunk 5 above)
Comment (franciscomunoz) [reply boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: done

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst (hunk 6 above)
Comment (franciscomunoz) [reply boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: done

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst (hunk 7 above)
Comment (franciscomunoz) [reply boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: done

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst (hunk 8 above)
Comment (franciscomunoz) [reply boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: done

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst (hunk 9 above)
Comment (franciscomunoz) [reply boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: done

File: boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst (hunk 10 above)
Comment (franciscomunoz) [reply boards/arm/mec2016evb_assy6797/doc/mec2016evb_assy6797.rst]: done

**Comment by nashif** [comment]
@franciscomunoz CI is failing on many tests, can you please take a look?
```
[10/122] Building C object CMakeFiles/app.dir/src/kernel.c.obj
[11/122] Building C object CMakeFiles/app.dir/src/event_flags.c.obj
FAILED: ccache /opt/sdk/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arm-zephyr-eabi/arm-zephyr-eabi-gcc -DBUILD_VERSION=zephyr-v1.13.0-4313-g7b7cf9b4d9 -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/include/cmsis_rtos_v2 -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/kernel/include -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/arch/arm/include -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/include -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/include/drivers -Izephyr/include/generated -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/soc/arm/microchip_mec/mec1701 -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/libc/minimal/include -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/cmsis/Include -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/microchip/mec -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/include -I/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/ztest/include -isystem /opt/sdk/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/lib/arm-zephyr-eabi/gcc/arm-zephyr-eabi/6.2.0/include -isystem /opt/sdk/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/lib/arm-zephyr-eabi/gcc/arm-zephyr-eabi/6.2.0/include-fixed -Os -g -Wall -Wformat -Wformat-security -Wno-format-zero-length -imacros /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/sanity-out/mec2016evb_assy6797/tests/cmsis_rtos_v2/portability.cmsis_rtos_v2/zephyr/include/generated/autoconf.h -ffreestanding -Wno-main -fno-common --sysroot /opt/sdk/zephyr-sdk-0.9.5/sysroots/armv5-zephyr-eabi/usr -mthumb -mcpu=cortex-m4 -Werror -Wno-deprecated-declarations -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -Wno-pointer-sign -Wno-unused-but-set-variable -fno-reorder-functions -fno-defer-pop -Werror=implicit-int -Wpointer-arith -ffunction-sections -fdata-sections -mabi=aapcs -march=armv7e-m -std=c99 -MD -MT CMakeFiles/app.dir/src/event_flags.c.obj -MF CMakeFiles/app.dir/src/event_flags.c.obj.d -o CMakeFiles/app.dir/src/event_flags.c.obj   -c /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/cmsis_rtos_v2/src/event_flags.c
In file included from /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/ztest/include/ztest_assert.h:19:0,
                 from /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/ztest/include/ztest.h:59,
                 from /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/cmsis_rtos_v2/src/event_flags.c:7:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/lib/libc/minimal/include/stdio.h:27:15: error: expected identifier or ( before - token
 #define EOF  (-1)
               ^
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/microchip/mec/MCHP_MEC1701.h:25480:22: note: in expansion of macro EOF
       __IO uint8_t   EOF        :  1;               /*!< [1..1] EOF (End of Frame) is asserted following Message Stop
                      ^~~

```


see https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/33634/2/tests

**Review by dbkinder — APPROVED** [review]
+1 for doc changes, thanks!

**Comment by dbkinder** [comment]
retest

**Comment by nashif** [comment]
@franciscomunoz there are still issues found with sanitycheck

**Comment by nashif** [comment]
note: license check verified locally, works fine, results are also available onlone at https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/13117/scancode.html

**Pull request merged**